analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -74,7 +74,12 @@ public void Execute()
                 return;
             }
             var current = queue.Dequeue();
-            if (visited.Add(current) && current.AddVisit() <= MaxOperationVisits)
+            if (visited.Add(current)
```

Comment:
What is the performance impact of this change?

Reply:
Two things:
- A few more branches will escape loops than before
- We call `IsInLoopCondition` a few more times
I'm not sure what that means in performance but it should be compensated by two follow-up PRs that will a) remove the "wrong" branch leaving the loop and b) remove calls to `IsInLoopCondition` during the operation handling.

Reply:
We need numbers. What happens when you analyze some of the ITs several times in average? Before vs. after

Reply:
@pavel-mikula-sonarsource  https://docs.google.com/spreadsheets/d/1sZxX2uzYnDjFTbGAZm9dKUNqkDgJj48VzadNzYq41ws/edit#gid=0

There's some increase - but IMO it's expected.

Reply:
@pavel-mikula-sonarsource  Updated - the numbers seem unstable + sometimes the percentage is +74% but it's worth seeing the numbers too - as it's from 1 second to 1.75 second which is not a big difference.

Reply:
Any row below 30s doesn't bring value, 74% of 1sec operation is random variance.

Reply:
Can you update the table? To delete the useless data and have only data that tell something? I still don't understand where we stand.

Reply:
Please check the two green columns at the bottom, they reflect the difference in the average run times between the master and the branch with the loop improvements - I deleted the percentages as they are misleading.

Negative means the loop improvements branch is faster.

Seems that the hit is not big except for two projects that we have +8 seconds - but not sure how significant this is.

Reply:
I have added some sums. I don't feel like this data is statistically sound. There are run-tim variations from over 1:40 in the same build. If we want to actually compare them, we need to run them much more often.

Reply:
The data still doesn't make any sense to me. What does positive `59:59.95` mean? That we are 59 minutes slower on something that took 0.1 sec before?

I suggest throwing it all away. Run akka.net from ITs 5x on master, 5x on the branch and compare the time.

Reply:
@pavel-mikula-sonarsource Apologies, I'm not familiar at all with google sheets. Data were converted to time instead of staying as duration.

Now it's fixed.

I won't delete them - now they make sense although they are indeed statistically insignificant. 

I'll do also the test with Akka and come back with the results.

Reply:
@pavel-mikula-sonarsource  https://docs.google.com/spreadsheets/d/1sZxX2uzYnDjFTbGAZm9dKUNqkDgJj48VzadNzYq41ws/edit#gid=979078328

Here's Akka run 6 times with master and the loop improvements. 
Hope this makes sense.

Reply:
Ok, looks good